### PR TITLE
Issue17

### PR DESCRIPTION
**Pull request by shua1090**
Example request (missing authentication)
{
    "userid": "01y5hf7YqGZQfzqlgDXj",
    "email": "hello!",
    "name": "shynn.lawrence@gmail.com",
    "phone": "9495724699",
    "instagram": "shua1008",
    "discord": "shua1008",
    "spotify": "shua1008",
    "snapchat": "N/A"
}

### CODE REVIEW

File: database/index.ts [hunk 1]
```diff
@@ -22,15 +23,82 @@ const firebaseConfig = {
...
+      "firstName": userData["firstName"],
+      "lastName": userData["lastName"],
+      "session": userData["session"],
+      "contacts": contacts,
+    });
+    console.log(`Updated with ${contacts}`);
+    return true;
+  }
+
+  async addContacts(id: String, contactID: String){
```
Comment (OlexG) [inline database/index.ts:63]: I assume this should be deleted

File: database/index.ts [hunk 2]
```diff
@@ -22,15 +23,82 @@ const firebaseConfig = {
...
+export const app = !getApps().length ? initializeApp(firebaseConfig) : getApps()[0];
+export const db = getFirestore(app);
 
 class UserDBManager {
   private db: Firestore;
   constructor(db: Firestore) {
     this.db = db;
   }
 
+  async addContactToID(id: String, contactID: String){
```
Comment (OlexG) [inline database/index.ts:35]: We probably don't need this either

**Review by OlexG — CHANGES_REQUESTED** [review]

File: database/index.ts (hunk 1 above)
Comment (shua1090) [reply database/index.ts:63]: deleted

File: database/index.ts (hunk 2 above)
Comment (shua1090) [reply database/index.ts:35]: deleted

File: pages/api/types/contact.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+type contact = {
```
[the remark this answers is not in the document]
Comment (shua1090) [reply pages/api/types/contact.ts:1]: nah lol we'll use id as same for userid and contactid, but contacts will have an array of contacts for each userid (contacts collection is userid-contacts[])

File: pages/api/addContact.ts
Comment (jeremiahliao) [inline pages/api/addContact.ts:1]: for adding contacts on front end and backend, we need to authenticate user both times correct?

**Review by jeremiahliao — CHANGES_REQUESTED** [review]
add rest of the parameters to contact object

File: pages/api/addContact.ts
Comment (shua1090) [reply pages/api/addContact.ts:1]: yeah I think so; changed parameters now btw!

File: database/index.ts [hunk 4]
```diff
@@ -22,15 +23,82 @@ const firebaseConfig = {
...
+      console.log("Adding");
+      await setDoc(doc(this.db, "contacts", q[0].id), { 
+        "phone": "shynn",
+        "test": "test"
+      });
+      console.log("added");
+    }
+  }
+
+  async getContactByPhone(field: string, value: BigInt){
```
[the remark this answers is not in the document]
Comment (shua1090) [reply database/index.ts]: added

**Comment by shua1090** [comment]
Example request:
{
    "userid": "01y5hf7YqGZQfzqlgDXj",
    "college": "Cal Poly SLO",
    "major": "Computer Science",
    "firstName": "Brian",
    "lastName": "Mai",
    "github": "reakunen"
}
Any non-filled field will be nulled automatically (shouldn't be a problem considering frontend would handle that for us)

**Review by jeremiahliao — APPROVED** [review]
looks good!

**Review by OlexG — APPROVED** [review]

**Comment by OlexG** [comment]
LGMT

**Pull request merged**